ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -464,34 +522,54 @@ forkBlockProduction IS{..} =
   New network layer
 -------------------------------------------------------------------------------}
 
-data NetworkRequires m up blk hdr = NetworkRequires {
```

Comment:
I think we should revert this abstraction to its previous form so we can decouple creating the clients and servers from actually connecting them over channels.